### PR TITLE
[MIGRATION] Add note about authentication errors and server errors

### PR DESCRIPTION
**Pull request by mojito317**
## Checklist

- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](../blob/master/DCO1.1.txt)
- [ ] Added tests for code changes _or_ test/build only changes
- [ ] Updated the change log file (`CHANGES.md`|`CHANGELOG.md`) _or_ test/build only changes
- [x] Completed the PR template below:

## Description
There have been some questions regarding migration lately. This PR tries to address them.

## Approach

I clarified some of the questionable topics in this change:
* where users should check the server errors and the authentication errors
* 
I also fixed a link for `Configuring the HTTP client section`

## Schema & API Changes

- "No change"

## Security and Privacy

- "No change"

## Testing

- N/A

## Monitoring and Logging

- "No change"

### CODE REVIEW

File: MIGRATION.md [hunk 1]
```diff
@@ -23,7 +23,12 @@ There are several ways to create a client connection in `cloudant-java-sdk`:
 1. There is no built-in pagination support for views. Examples coming soon.
 1. Replay interceptors are replaced by the [automatic retries](https://github.com/IBM/ibm-cloud-sdk-common/#automatic-retries) feature for failed requests.
 1. Error handling is not transferable from `java-cloudant` to `cloudant-java-sdk`. For more information go to the [Error handling section](https://cloud.ibm.com/apidocs/cloudant?code=java#error-handling) in our API docs.
-1. Custom HTTP client configurations in `java-cloudant` are not transferable to `cloudant-java-sdk`. For more information go to the [Configuring the HTTP client section(https://githubcom/IBM/ibm-cloud-sdk-common/#configuring-the-http-client) in the IBM Cloud SDK Common README.
+1. Custom HTTP client configurations in `java-cloudant` are not transferable to 
+   `cloudant-java-sdk`. For more information go to the [Configuring the HTTP client section](https://githubcom/IBM/ibm-cloud-sdk-common/#configuring-the-http-client) in the IBM Cloud SDK Common README.
```
Comment (vmatyusGitHub) [inline MIGRATION.md:27]:
```suggestion
   `cloudant-java-sdk`. For more information go to the [Configuring the HTTP client section](https://github.com/IBM/ibm-cloud-sdk-common/#configuring-the-http-client) in the IBM Cloud SDK Common README.
```

Comment (mojito317) [reply MIGRATION.md]: Done in a389d6f

File: MIGRATION.md [hunk 2]
```diff
@@ -23,7 +23,13 @@ There are several ways to create a client connection in `cloudant-java-sdk`:
 1. There is no built-in pagination support for views. Examples coming soon.
 1. Replay interceptors are replaced by the [automatic retries](https://github.com/IBM/ibm-cloud-sdk-common/#automatic-retries) feature for failed requests.
 1. Error handling is not transferable from `java-cloudant` to `cloudant-java-sdk`. For more information go to the [Error handling section](https://cloud.ibm.com/apidocs/cloudant?code=java#error-handling) in our API docs.
-1. Custom HTTP client configurations in `java-cloudant` are not transferable to `cloudant-java-sdk`. For more information go to the [Configuring the HTTP client section(https://githubcom/IBM/ibm-cloud-sdk-common/#configuring-the-http-client) in the IBM Cloud SDK Common README.
+1. Custom HTTP client configurations in `java-cloudant` are not transferable to 
+   `cloudant-java-sdk`. For more information go to the [Configuring the HTTP client section]
+   (https://github.com/IBM/ibm-cloud-sdk-common/#configuring-the-http-client) in the IBM Cloud SDK Common README.
+1. Authentication errors turn out at the time of instantiation of a client, while errors
```
Comment (emlaver) [inline MIGRATION.md:29]: I think we should split this into two items.
Wdyt about these changes:
```suggestion
1. Authentication errors occur during service instantiation.  For example, the code `Cloudant service = Cloudant.newInstance("EXAMPLE");` will fail with `Authentication information was not properly configured.` if required environment variables prefixed with `EXAMPLE` are not set.
1. Server errors occur when running a request against the service.  We suggest to
   check server errors with [`getServerInformation`](https://cloud.ibm.com/apidocs/cloudant?
    code=java#getserverinformation) which is the new alternative of `metaInformation()`.
```

Comment (mojito317) [reply MIGRATION.md]: Done in a91600a

**Review by vmatyusGitHub — APPROVED** [review]
LGTM :+1:

**Review by emlaver — APPROVED** [review]

**Comment by mojito317** [comment]
Thanks for the reviews @emlaver, @vmatyusGitHub! I am merging this!

**Pull request merged**